sdk/monitor/query/azlogs/client_test.go

```diff
@@ -205,6 +205,103 @@ func TestQueryResource_Advanced(t *testing.T) {
 	testSerde(t, &res)
 }
 
+func TestQueryBatch_QuerySuccess(t *testing.T) {
+	client := startTest(t)
+	query1, query2 := query, query+" | take 2"
+	timespan := azlogs.NewTimeInterval(time.Date(2022, 3, 2, 0, 0, 0, 0, time.UTC), time.Date(2022, 3, 3, 0, 0, 0, 0, time.UTC))
+
+	batchRequest := azlogs.BatchRequest{[]azlogs.BatchQueryRequest{
+		{Body: &azlogs.QueryBody{Query: to.Ptr(query1), Timespan: to.Ptr(timespan)}, ID: to.Ptr("1"), WorkspaceID: to.Ptr(workspaceID)},
+		{Body: &azlogs.QueryBody{Query: to.Ptr(query2), Timespan: to.Ptr(timespan)}, ID: to.Ptr("2"), WorkspaceID: to.Ptr(workspaceID)},
+	}}
+	testSerde(t, &batchRequest)
+
+	res, err := client.QueryBatch(context.Background(), batchRequest, nil)
+	require.NoError(t, err)
+	require.Len(t, res.Responses, 2)
+	for _, resp := range res.Responses {
+		require.Nil(t, resp.Body.Error)
+		require.NotNil(t, resp.Body.Tables)
+		if *resp.ID == "1" && len(resp.Body.Tables[0].Rows) != 100 {
+			t.Fatal("expected 100 rows from batch request 1")
+		}
+		if *resp.ID == "2" && len(resp.Body.Tables[0].Rows) != 2 {
+			t.Fatal("expected 2 rows from batch request 2")
+		}
```

Comment:
Where does this data come from? I'm wondering whether these tests depend on the state of resources they don't control.

Reply:
It all in the kusto query in line 18! The query provides the data, so it's always constant. No test resources required.